### PR TITLE
[mongo] report dbtop metrics

### PR DESCRIPTION
**Pull request by scottbessler**
The output of dbtop from the mongo server web interface is very useful for debugging performance issues. It is available as the `top` admin command and combined with tags makes for a great experience with datadog.

I'm using it here:
mongodb datadog 2015-09-11 09-49-30

### CODE REVIEW

**Comment by remh** [comment]
Thanks a lot @scottbessler we'll review this for our 5.6.0 release!

**Comment by benmccann** [comment]
@scottbessler can you rebase this PR?

**Comment by scottbessler** [comment]
rebased.

**Comment by benmccann** [comment]
@scottbessler i'm afraid this one requires another rebase. one of the other mongodb PRs was just merged that touches this file. thanks!

**Comment by scottbessler** [comment]
rebased.

**Comment by benmccann** [comment]
@remh @yannmh this one looks good to me. only question is if you want to put it behind a flag like you've been doing with the other metrics?

File: checks.d/mongo.py [hunk 1]
```diff
@@ -448,3 +448,37 @@ def check(self, instance):
...
+            for ns, ns_metrics in dbtop['totals'].iteritems():
+                if "." not in ns:
+                    continue
+
+                # configure tags for db name and collection name
+                dbname, collname = ns.split(".", 1)
+                ns_tags = tags + ["db:%s" % dbname, "collection:%s" % collname]
+
+                # iterate over DBTOP metrics
+                for m in self.DBTOP:
```
Comment (yannmh) [inline checks.d/mongo.py]: Did `self.DBTOP` get lost during the rebases ? :mag:

**Comment by irabinovitch** [comment]
The failing tests here are from flake8:

./checks.d/mongo.py:476:25: F821 undefined name 'types'
./checks.d/mongo.py:476:40: F821 undefined name 'types'
./checks.d/mongo.py:476:56: F821 undefined name 'types'
./checks.d/mongo.py:484:1: W391 blank line at end of file

**Comment by irabinovitch** [comment]
@yannmh  Any thoughts on @benmccann 's [question](https://github.com/DataDog/dd-agent/pull/1906#issuecomment-152758130) here?

**Comment by benmccann** [comment]
@scottbessler could you fix the flake8 warnings?

@yannmh any thoughts on whether this needs to be behind a flag?

**Comment by irabinovitch** [comment]
@yannmh  Just a friendly ping on @benmccann 's questions here.

@scottbessler  can you rebase when you have a moment the merge window for 5.7 is closing shortly.

**Comment by scottbessler** [comment]
@irabinovitch rebased, flake8 fixed

**Comment by yannmh** [comment]
Thanks for your contribution @scottbessler. 

The number of metrics introduced is reasonably small, but provides very interesting use case (as the one provided) to monitor MongoDB instances. Thus, if it's not slowing down the check, let's collect these metrics by default.

However, I think it would be wiser to postpone these changes for the 5.8.0 Datadog Agent release. The code in its current state has a LOT of redundancy, and a very poor unit and integration test coverage. I'd rather fix this, before merging additional changes.

**Comment by irabinovitch** [comment]
@yannmh Can we provide feedback on the specific issues in line so that Scott or others can address?

**Comment by yannmh** [comment]
Sorry for the confusion: my comments actually concern the check in its current state but not the changes brought. 

Also, on a second thought, let's merge these changes this week: the code refactoring and testing I mentioned, have no reason to delay it. I'll handle these extra changes on a future separate pull request.

**Comment by irabinovitch** [comment]
Thanks @yannmh. Appreciate the clarification.  Should I file an issue to track the mongodb check refactor/clean up?

**Comment by yannmh** [comment]
Rebased and reopened on #2166. Thanks again @scottbessler.

**Pull request closed without merging**